generic_client.go

```diff
@@ -5,28 +5,31 @@ import (
 	"encoding/json"
 	"fmt"
 	"io"
-	"io/ioutil"
 	"net/http"
 	"net/url"
 	"regexp"
 	"strings"
+	"time"
+
+	"github.com/opentable/sous/util/logging"
+	"github.com/opentable/sous/util/logging/messages"
 )
 
 type (
 	// Requester defines the interface that Swaggering uses to
 	// make actual HTTP requests of the API server
 	Requester interface {
 		// DTORequest performs an HTTP request and populates a DTO based on the response
-		DTORequest(dto DTO, method, path string, pathParams, queryParams urlParams, body ...DTO) error
+		DTORequest(resourceName string, dto DTO, method, path string, pathParams, queryParams urlParams, body ...DTO) error
 
 		// Request performs an HTTP request and returns the body of the response
-		Request(method, path string, pathParams, queryParams urlParams, body ...DTO) (io.ReadCloser, error)
+		Request(resourceName string, method string, path string, pathParams urlParams, queryParams urlParams, body ...DTO) (io.ReadCloser, error)
 	}
 
 	// GenericClient is a generic client for Swagger described services
 	GenericClient struct {
 		BaseURL string
-		Logger  Logger
+		Logger  logging.LogSink
```

Comment:
If instead `Logger` was `LogFunc(http.Response, string, time.Duration)` we could avoid the coupling to Sous entirely by having Sous pass in that func pre-configured to use our `logging.LogSink`. Just a thought, I know this was done for speed of implementation which I'm all for.